File: BadCode/Palindrome.py
# Bad Code

def is_palindrome(s):
    start = 0
    end = len(s) - 1

    while start < end:
        if s[start] != s[end]:
            return False
        start += 1
        end -= 1

    return True


input_str = "racecar"
is_pal = is_palindrome(input_str)
print(f"Is '{input_str}' a palindrome? {is_pal}")

File: BadCode/Palindrome.java
package BadCode;

public class Palindrome {
    public static boolean isPalindrome(String s) {
        int start = 0;
        int end = s.length() - 1;

        while (start < end) {
            if (s.charAt(start) != s.charAt(end)) {
                return false;
            }
            start++;
            end--;
        }

        return true;
    }

    public static void main(String[] args) {
        String input = "racecar";
        boolean isPal = isPalindrome(input);
        System.out.println("Is '" + input + "' a palindrome? " + isPal);
    }
}
